src/routes/safe/components/Transactions/helpers/SpendingLimitModalWrapper/index.tsx

```diff
@@ -0,0 +1,203 @@
+import { TxEstimatedFeesDetail } from 'src/routes/safe/components/Transactions/helpers/TxEstimatedFeesDetail'
+import { useEstimationStatus } from 'src/logic/hooks/useEstimationStatus'
+import { Errors, logError } from 'src/logic/exceptions/CodedException'
+import { ButtonStatus, Modal } from 'src/components/Modal'
+import { lg, md } from 'src/theme/variables'
+import { Operation } from '@gnosis.pm/safe-react-gateway-sdk'
+import { getNativeCurrency } from 'src/config'
+import useSafeAddress from 'src/logic/currentSession/hooks/useSafeAddress'
+import { sameAddress, ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
+import { useEstimateGasLimit } from 'src/logic/hooks/useEstimateGasLimit'
+import { useExecutionStatus } from 'src/logic/hooks/useExecutionStatus'
+import {
+  AllowanceTransferProps,
+  checkAllowanceTransferExecution,
+  estimateGasForAllowanceTransfer,
+} from 'src/logic/safe/transactions/allowance'
+import { toTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+import { EMPTY_DATA } from 'src/logic/wallets/ethTransactions'
+import { Token } from 'src/logic/tokens/store/model/token'
+import { getNativeCurrencyAddress } from 'src/config/utils'
+
+type Props = {
+  children: ReactNode
+  operation?: Operation
+  txNonce?: string
+  txData: string
+  txValue?: string
+  txTo?: string
+  txType?: string
+  onSubmit: (txParams: TxParameters) => void
+  onClose?: () => void
+  onBack?: (...rest: any) => void
+  submitText?: string
+  isSubmitDisabled?: boolean
+  txToken?: Token
+  txDelegate?: string
+  txAmount?: string
+}
+
+const Container = styled.div`
+  padding: 0 ${lg} ${md};
+`
+
+export const SpendingLimitModalWrapper = ({
+  children,
+  txNonce,
+  txTo,
+  txType,
+  onSubmit,
+  onBack,
+  onClose,
+  submitText,
+  isSubmitDisabled,
+  txToken,
+  txDelegate,
+  txAmount,
+}: Props): React.ReactElement => {
+  const [manualGasPrice, setManualGasPrice] = useState<string>()
+  const [manualMaxPrioFee, setManualMaxPrioFee] = useState<string>()
+  const [manualGasLimit, setManualGasLimit] = useState<string>()
+  const isSendingNativeToken = useMemo(() => sameAddress(txToken?.address, getNativeCurrencyAddress()), [txToken])
+  const { safeAddress } = useSafeAddress()
+  const nativeCurrency = getNativeCurrency()
+
+  const allowanceTransferParams = useMemo((): AllowanceTransferProps => {
+    return {
+      safe: safeAddress,
+      token: isSendingNativeToken ? ZERO_ADDRESS : txToken?.address || ZERO_ADDRESS,
+      to: txTo || ZERO_ADDRESS,
+      amount: toTokenUnit(txAmount || '0', txToken?.decimals || 0),
+      paymentToken: ZERO_ADDRESS,
+      payment: 0,
+      delegate: txDelegate || ZERO_ADDRESS,
+      signature: EMPTY_DATA,
+    }
+  }, [isSendingNativeToken, safeAddress, txAmount, txDelegate, txTo, txToken?.address, txToken?.decimals])
+
+  const gasLimit = useEstimateGasLimit(
+    () => estimateGasForAllowanceTransfer(allowanceTransferParams),
```

Comment:
Instead of passing a callback, I would pass a flag isSpendingLimit and let the hook pick the right function. Otherwise it's a bit too DIY.

Reply:
I thought about that too but it would require us to pass `allowanceTransferParams` and `txParameters` into both hooks and use them conditionally there. The current structure makes it really easy to test the logic of both hooks since we can just mock the callback functions without having to worry about the huge number of parameters.